main.tf

```
@@ -0,0 +1,279 @@
+      dynamic "rule_variables" {
+        for_each = lookup(each.value.rule_group, "rule_variables", null) != null ? [true] : []
+        content {
+          # Set of configuration blocks that define IP address information
+          dynamic "ip_sets" {
+            for_each = lookup(each.value.rule_group.rule_variables, "ip_sets", [])
+            content {
+              key = ip_sets.value.key
+              ip_set {
+                definition = ip_sets.value.definition
+              }
+            }
+          }
+          # Set of configuration blocks that define port range information
+          dynamic "port_sets" {
+            for_each = lookup(each.value.rule_group.rule_variables, "port_sets", [])
+            content {
+              key = port_sets.value.key
+              port_set {
+                definition = port_sets.value.definition
+              }
+            }
+          }
+        }
+      }
+
+      # `stateful_rule_options` - a configuration block that defines stateful rule options for the rule group
+      # If the STRICT_ORDER rule order is specified, this rule group can only be referenced in firewall policies that also utilize STRICT_ORDER for the stateful engine
+      # STRICT_ORDER can only be specified when using a rules_source of rules_string or stateful_rule
+      dynamic "stateful_rule_options" {
+        for_each = var.rule_group_stateful_engine_options_rule_order != null && var.rule_group_stateful_engine_options_rule_order != "" ? [true] : []
+        content {
+          # Indicates how to manage the order of the rule evaluation for the rule group
+          # Default value: DEFAULT_ACTION_ORDER
+          # Valid values: DEFAULT_ACTION_ORDER, STRICT_ORDER
+          rule_order = var.rule_group_stateful_engine_options_rule_order
+        }
+      }
+
+      # `rules_source` - a configuration block that defines the stateful or stateless rules for the rule group
+      # Only one of `rules_source_list`, `rules_string`, `stateful_rule`, or `stateless_rules_and_custom_actions` must be specified
+      rules_source {
+        rules_string = lookup(each.value.rule_group.rules_source, "rules_string", null)
+
+        dynamic "rules_source_list" {
+          for_each = lookup(each.value.rule_group.rules_source, "rules_source_list", null) != null ? [true] : []
+          content {
+            # String value to specify whether domains in the target list are allowed or denied access. Valid values: ALLOWLIST, DENYLIST
+            generated_rules_type = each.value.rule_group.rules_source.rules_source_list.generated_rules_type
+            # Set of types of domain specifications that are provided in the targets argument. Valid values: HTTP_HOST, TLS_SNI
+            target_types = each.value.rule_group.rules_source.rules_source_list.target_types
+            # Set of domains that you want to inspect for in your traffic flows
+            targets = each.value.rule_group.rules_source.rules_source_list.targets
```

This method is more readable in my opinion

```suggestion
          for_each = lookup(each.value.rule_group.rules_source, "rules_source_list", {})
          content {
            # String value to specify whether domains in the target list are allowed or denied access. Valid values: ALLOWLIST, DENYLIST
            generated_rules_type = rules_source_list.value.generated_rules_type
            # Set of types of domain specifications that are provided in the targets argument. Valid values: HTTP_HOST, TLS_SNI
            target_types = rules_source_list.value.target_types
            # Set of domains that you want to inspect for in your traffic flows
            targets = rules_source_list.value.targets
```

This could be done for a lot of these dynamics. What do you think?